core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -257,10 +262,16 @@ public void startUp(Session session, String keyspace, boolean resetDb, boolean c
                 .put(STRING, Functions::getStringDataPoint)
                 .build();
 
+        metricsIdxCache = CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.HOURS).maximumSize(40000).build();
```

Comment:
The idea of this cache was to just avoid writing to the metrics index when there is a burst of activity for a single metric. That is why elements in the cache are evicted if there is no activity for a metric for a period of time.

Loading the cache at startup is not a good idea because of the memory implications. There can be a lot of metrics that were written once and never again to be written. I prefer this cache to be based on the current metrics activity.